tfx/components/transform/executor.py

```diff
@@ -276,29 +278,45 @@ def Do(self, input_dict: Dict[Text, List[types.Artifact]],
     Args:
       input_dict: Input dict from input key to a list of artifacts, including:
         - input_data: A list of type `standard_artifacts.Examples` which
-          should contain two splits 'train' and 'eval'.
+          should contain two splits 'train' and 'eval' if custom splits
+          is not specified in splits_config.
         - schema: A list of type `standard_artifacts.Schema` which should
           contain a single schema artifact.
       output_dict: Output dict from key to a list of artifacts, including:
         - transform_output: Output of 'tf.Transform', which includes an exported
           Tensorflow graph suitable for both training and serving;
         - transformed_examples: Materialized transformed examples, which
-          includes both 'train' and 'eval' splits.
-      exec_properties: A dict of execution properties, including either one of:
+          includes both 'train' and 'eval' splits if custom splits is not
+          specified in splits_config.
+      exec_properties: A dict of execution properties, including:
         - module_file: The file path to a python module file, from which the
           'preprocessing_fn' function will be loaded.
         - preprocessing_fn: The module path to a python function that
           implements 'preprocessing_fn'.
+        - splits_config: A transform_pb2.SplitsConfig instance, providing splits
+          that should be analyzed and splits that should be transformed. Default
+          behavior is analyze the 'train' split (when analyze_splits is not set)
+          and transform both 'train' and 'eval' splits (when transform_splits is
+          not set).
 
     Returns:
       None
     """
     self._log_startup(input_dict, output_dict, exec_properties)
 
-    train_data_uri = artifact_utils.get_split_uri(input_dict[EXAMPLES_KEY],
-                                                  'train')
-    eval_data_uri = artifact_utils.get_split_uri(input_dict[EXAMPLES_KEY],
-                                                 'eval')
+    splits_config = transform_pb2.SplitsConfig()
+    if exec_properties.get('splits_config', None):
+      json_format.Parse(exec_properties['splits_config'], splits_config)
+
+    if not splits_config.analyze_splits:
+      splits_config.analyze_splits.append('train')
+      logging.info("Analyze the 'train' split when splits_config.analyze_splits"
+                   " is not set.")
+    if not splits_config.transform_splits:
+      splits_config.transform_splits.extend(['train', 'eval'])
```

Comment:
should we extend all splits, not just train and eval?

actually I think the default behavior should be

if exec_properties['splits_config'] is None,
we create a SplitsConfig with analyze `train` and transform all split

if splits_config is set, we don't change what user set

Reply:
yes, I think it makes sense to transform all splits, I will make this change

Reply:
if splits_config is set empty by the user, I think we still want to change it to the default behavior. that's why i use if analyze_splits is empty to check for whether we want default behavior.

Reply:
because it's possible user just set analyze but not transform, so i checked splits_config.analyze_splits and splits_config.transform_splits separately. if we directly check for splits_config, if splits_config is not set correctly (maybe user sets analyze or transform to be empty), then i don't know which one i should add the default behavior.